libafl_qemu/librasan/asan/src/maps/decode.rs

```diff
@@ -0,0 +1,34 @@
+use core::fmt::Debug;
+
+use thiserror::Error;
+
+pub trait MapDecode<T> {
+    fn from_hex(c: u8) -> Result<T, MapDecodeError>;
+    fn from_dec(c: u8) -> Result<T, MapDecodeError>;
+}
+
+impl<T: From<u8>> MapDecode<T> for T {
+    fn from_hex(c: u8) -> Result<T, MapDecodeError> {
+        let c = match c {
+            b'0'..=b'9' => c - b'0',
+            b'a'..=b'f' => c - b'a' + 10,
```

Comment:
I _guess_ the two match branches could be or-red (but maybe not for range stuff?)

Reply:
I don't think you can for ranges anyway. But the right hand side of each is different in any case?